ddtrace/tracer/log.go

```diff
@@ -107,6 +109,13 @@ func logStartup(t *tracer) {
 	if _, err := samplingRulesFromEnv(); err != nil {
 		info.SamplingRulesError = fmt.Sprintf("%s", err)
 	}
+	// Based on the apply logic in sampler.go func (*rulesSampler).applyRate,
+	// the rate limit applies only when there are rules or there is a global rate set.
+	if t.rulesSampling.globalRate > 0 || len(t.rulesSampling.rules) > 0 {
+		if t.rulesSampling != nil && t.rulesSampling.limiter != nil && t.rulesSampling.limiter.limiter != nil {
```

Comment:
should the check `t.rulesSampling != nil` be moved up a line here? (otherwise wouldn't `t.rulesSampling.globalRate` crash if it's nil?